carma_cooperative_perception/include/carma_cooperative_perception/multiple_object_tracker_component.hpp

```diff
@@ -0,0 +1,84 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CARMA_COOPERATIVE_PERCEPTION__MULTIPLE_OBJECT_TRACKER_COMPONENT_HPP_
+#define CARMA_COOPERATIVE_PERCEPTION__MULTIPLE_OBJECT_TRACKER_COMPONENT_HPP_
+
+#include <carma_ros2_utils/carma_lifecycle_node.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <carma_cooperative_perception_interfaces/msg/detection_list.hpp>
+#include <carma_cooperative_perception_interfaces/msg/track_list.hpp>
+
+#include <cooperative_perception/ctra_model.hpp>
+#include <cooperative_perception/ctrv_model.hpp>
+#include <cooperative_perception/track_management.hpp>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+namespace carma_cooperative_perception
+{
+
+using Detection =
+  std::variant<cooperative_perception::CtrvDetection, cooperative_perception::CtraDetection>;
+using Track = std::variant<cooperative_perception::CtrvTrack, cooperative_perception::CtraTrack>;
+
+auto make_detection(const carma_cooperative_perception_interfaces::msg::Detection & msg)
+  -> Detection;
+
+class MultipleObjectTrackerNode : public carma_ros2_utils::CarmaLifecycleNode
+{
+public:
+  explicit MultipleObjectTrackerNode(const rclcpp::NodeOptions & options);
+
+  auto handle_on_configure(const rclcpp_lifecycle::State & /* previous_state */)
+    -> carma_ros2_utils::CallbackReturn override;
+
+  auto handle_on_activate(const rclcpp_lifecycle::State & /* previous_state */)
+    -> carma_ros2_utils::CallbackReturn override;
+
+  auto handle_on_deactivate(const rclcpp_lifecycle::State & /* previous_state */)
+    -> carma_ros2_utils::CallbackReturn override;
+
+  auto handle_on_cleanup(const rclcpp_lifecycle::State & /* previous_state */)
+    -> carma_ros2_utils::CallbackReturn override;
+
+  auto handle_on_shutdown(const rclcpp_lifecycle::State & /* previous_state */)
+    -> carma_ros2_utils::CallbackReturn override;
+
+  auto store_new_detections(
+    const carma_cooperative_perception_interfaces::msg::DetectionList & msg) noexcept -> void;
+
+  auto execute_pipeline() -> void;
+
+private:
+  rclcpp::Subscription<carma_cooperative_perception_interfaces::msg::DetectionList>::SharedPtr
+    detection_list_sub_{nullptr};
+
+  rclcpp_lifecycle::LifecyclePublisher<
+    carma_cooperative_perception_interfaces::msg::TrackList>::SharedPtr track_list_pub_{nullptr};
+
+  rclcpp::TimerBase::SharedPtr pipeline_execution_timer_{nullptr};
+
+  std::vector<Detection> detections_;
+  std::unordered_map<cooperative_perception::Uuid, std::size_t> uuid_index_map_;
+  cooperative_perception::FixedThresholdTrackManager<Track> track_manager_;
+  units::time::nanosecond_t execution_period_{1 / units::frequency::hertz_t{2.0}};
```

Comment:
This is modifiable as a ROS 2 parameter. Do we need an additional change?

Reply:
Yeah I was thinking of the yaml file. We make volume to that yaml file, so that we don't have to set it runtime, but be bale to configure it before it starts without using the default value.

Reply:
A `params.yaml` file that we pass use in the launch configuration, or are you talking about a different one?

Reply:
something like this for example
https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/carma_wm_ctrl/config/parameters.yaml